src/promptflow/promptflow/executor/_async_nodes_scheduler.py

```diff
@@ -76,7 +99,30 @@ async def execute(
         # This is because it will always call `executor.shutdown()` when exiting the `with` block.
         # Then the event loop will wait for all tasks to be completed before raising the cancellation error.
         # See reference: https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor
-        outputs = await self._execute_with_thread_pool(executor, nodes, inputs, context)
+        try:
+            if line_timeout_sec is not None:
+                outputs = await self._execute_with_timeout(executor, nodes, inputs, context, line_timeout_sec)
+            else:
+                outputs = await self._execute_with_thread_pool(executor, nodes, inputs, context)
```

Comment:
Please rename this function to `self._execute`

You see that this is weird that:
```
if line_timeout_sec is not None:
    self._execute_with_timeout
else:
    self._execute_with_thread_pool
```
But this can be another PR

Reply:
Changed to self._execute.